app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -26,41 +34,72 @@ public void setJedisOps(JedisOps jedisOps) {
         this.jedisOps = jedisOps;
     }
 
-    public void setUserSession(String key, UserSession session) {
+    public void setUserSession(final UserSession session) {
+        checkNotNull(session);
+        List<Object> results = null;
         try {
-            String ser = BridgeObjectMapper.get().writeValueAsString(session);
-            String redisKey = RedisKey.SESSION.getRedisKey(key);
-            String result = jedisOps.setex(redisKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, ser);
+            final String userId = session.getUser().getId();
+            final String userKey = RedisKey.USER_SESSION.getRedisKey(userId);
+            final String sessionToken = session.getSessionToken();
+            final String sessionKey = RedisKey.SESSION.getRedisKey(sessionToken);
+            final String ser = BridgeObjectMapper.get().writeValueAsString(session);
+            results = jedisOps.getTransaction()
+                    .setex(userKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, sessionToken)
```

Comment:
This returns the session by GUID, unless you've changed it to look up the session by the user's ID. As a consequence, I can sign in multiple times without re-submittin the GUID I got from my other sessions. Then when deleted/signed out, only one of those sessions will be destroyed (the one whose GUID was submitted for the deletion or signing out). If getSessionIfItExists() is first looking up the user's ID from the GUID, then this should no longer happen, problem solved.